src/storage/btree.c

```diff
@@ -12689,57 +12769,92 @@ btree_node_common_prefix (THREAD_ENTRY * thread_p, BTID_INT * btid, PAGE_PTR pag
   LEAF_REC leaf_pnt;
   int error = NO_ERROR;
 
-  if (btree_node_is_compressed (thread_p, btid, page_ptr) == false)
+  if (TP_DOMAIN_TYPE (btid->key_type) != DB_TYPE_MIDXKEY)
     {
-      return 0;
+      /* Compression is only possible for indexes of DB_TYPE_MIDXKEY. */
+      goto not_compressed;
+    }
+
+  key_cnt = btree_node_number_of_keys (thread_p, page_ptr);
+  if (key_cnt < 2)
+    {
+      goto not_compressed;
+    }
+
+  header = btree_get_node_header (thread_p, page_ptr);
+  if (header == NULL)
+    {
+      assert (false);
+      goto not_compressed;
+    }
+
+  if (header->node_level > 1)
+    {
+      /* Compression is only possible at leaf nodes. */
+      goto not_compressed;
```

Comment:
Nit:

The code in this part is identical to the part of btree_node_get_common_prefix(). How about writing it as a separate function and using it?

Reply:
Thank you.
As you suggested, I thought about whether it would be better to make this common code into separate functions.
```
```

I left these codes so that there would be no difference with AS-IS.
I wanted to minimize the code that needed verification in this PR.

I think you should have checked this code before calling this function.
If I had to change anything further here, I think it would be to make sure that these codes are checked before this function is called.
I don't think there's a need to make it a separate function.